File: security/src/main/java/top/bettercode/summer/security/support/SecurityParameterNames.java
package top.bettercode.summer.security.support;

public interface SecurityParameterNames {

  /**
   * {@code grant_type} - used in Access Token Request.
   */
  String GRANT_TYPE = "grant_type";

  /**
   * {@code scope} - used in Authorization Request, Authorization Response, Access Token Request and
   * Access Token Response.
   */
  String SCOPE = "scope";

  /**
   * {@code access_token} - used in Authorization Response and Access Token Response.
   */
  String ACCESS_TOKEN = "access_token";

  /**
   * 兼容toekn名称
   */
  String COMPATIBLE_ACCESS_TOKEN = "accessToken";

  /**
   * {@code refresh_token} - used in Access Token Request and Access Token Response.
   */
  String REFRESH_TOKEN = "refresh_token";

  /**
   * {@code username} - used in Access Token Request.
   */
  String USERNAME = "username";

	/**
	 * {@code password} - used in Access Token Request.
	 */
	String PWDNAME = "password";

}

File: security/src/main/kotlin/top/bettercode/summer/security/support/SecurityParameterNames.kt
package top.bettercode.summer.security.support

interface SecurityParameterNames {
    companion object {
        /**
         * `grant_type` - used in Access Token Request.
         */
        const val GRANT_TYPE = "grant_type"

        /**
         * `scope` - used in Authorization Request, Authorization Response, Access Token Request and
         * Access Token Response.
         */
        const val SCOPE = "scope"

        /**
         * `access_token` - used in Authorization Response and Access Token Response.
         */
        const val ACCESS_TOKEN = "access_token"

        /**
         * 兼容toekn名称
         */
        const val COMPATIBLE_ACCESS_TOKEN = "accessToken"

        /**
         * `refresh_token` - used in Access Token Request and Access Token Response.
         */
        const val REFRESH_TOKEN = "refresh_token"

        /**
         * `username` - used in Access Token Request.
         */
        const val USERNAME = "username"

        /**
         * `password` - used in Access Token Request.
         */
        const val PWDNAME = "password"

        /**
         * revoke token
         */
        const val REVOKE_TOKEN = "revoke_token"
    }
}
